sway/desktop/output.c

```diff
@@ -577,10 +577,11 @@ static void handle_request_state(struct wl_listener *listener, void *data) {
 	assert(committed == 0);
 	store_output_config(oc);
 
-	wlr_scene_optimized_blur_set_size(output->layers.blur_layer,
-			oc->width, oc->height);
-
 	force_modeset();
+
+	int output_width, output_height;
+	wlr_output_transformed_resolution(output->wlr_output, &output_width, &output_height);
```

Comment:
Should this happen in scene? Feels weird we handle other node rotations there

Reply:
Fair enough. Posting a pr :)

Reply:
Hmm, not sure if that's the best idea, we don't change the size of the other nodes, even if presented on a rotated output. The optimized blur node could also be used for other cases than one that follows the output size and rotation.

Reply:
> Hmm, not sure if that's the best idea, we don't change the size of the other nodes, even if presented on a rotated output. The optimized blur node could also be used for other cases than one that follows the output size and rotation.

Or maybe we do do that to other nodes? I'll double check tomorrow if you haven't already :)

Reply:
Ok, so we do actually use the transformed size in scenefx, which is the culprit to the issues. The size has to be it's non-transformed values when used in this use-case. So the transformed size is needed to undo the transformation.